core/ledger/kvledger/txmgmt/txmgr/tx_simulator.go

```diff
@@ -36,6 +40,42 @@ func (s *txSimulator) SetState(ns string, key string, value []byte) error {
 		return err
 	}
 	s.rwsetBuilder.AddToWriteSet(ns, key, value)
+	// if this has a key level signature policy, add it to the interest
+	return s.checkKeySignaturePolicy(ns, key)
+}
+
+// If this key has a SBE policy, add that policy to the set
+func (s *txSimulator) checkKeySignaturePolicy(ns string, key string) error {
+	var metadata []byte
+	var err error
+	if metadata, err = s.txmgr.db.GetStateMetadata(ns, key); err != nil {
+		return err
+	}
+	return s.policyFromMetadata(metadata)
+}
+
+// If this private collection key has a SBE policy, add that policy to the set
+func (s *txSimulator) checkPrivateKeySignaturePolicy(ns string, coll string, key string) error {
+	var metadata []byte
+	var err error
+	var versionedValue *statedb.VersionedValue
+	if versionedValue, err = s.txmgr.db.GetPrivateDataHash(ns, coll, key); err != nil {
+		return err
+	}
+	_, metadata, _ = decomposeVersionedValue(versionedValue)
```

Comment:
```suggestion
s.txmgr.db.GetPrivateDataMetadataByHash(ns, coll, util.ComputeStringHash(key))
```
like `GetStateMetadata` in the case of public key, `GetPrivateDataMetadataByHash` applies an [optimization](https://github.com/hyperledger/fabric/blob/9a922fd9c78c01d185bc2b45fb98012a37f7bfb9/core/ledger/kvledger/txmgmt/privacyenabledstate/db.go#L285-L287) that internally skips looking up for the keys for a chaincode that never used SBE